File: ptr.go
package main

import "fmt"

// Function to increment an integer using a pointer
func increment(num *int) {
	*num++ // Increment the value pointed to by 'num'
}

func main() {
	value1 := 5
	value2 := 10

	addr1 := &value1 // Get the address of 'value1'

	fmt.Printf("Before increment: value1 = %d, value2 = %d\n", *addr1, value2)

	// Call the increment function to increment 'value1' and 'value2'
	increment(addr1) // Pass 'addr1' to the increment function for 'value1'
	increment(&value2) // Pass the address of 'value2' directly to the increment function

	fmt.Printf("After increment: value1 = %d, value2 = %d\n", *addr1, value2)
}

File: ptr.cpp
#include <iostream>

// Function to increment an integer using a pointer
void increment(int &num) {
    num++; // Increment the value directly
}

int main() {
    int value1 = 5;
    int value2 = 10;

    int &ref1 = value1; // Create a reference to 'value1'
    
    std::cout << "Before increment: value1 = " << ref1 << ", value2 = " << value2 << std::endl;

    // Call the increment function to increment 'value1' and 'value2'
    increment(ref1); // Pass the reference to 'value1' to the increment function
    increment(value2); // Pass 'value2' directly to the increment function

    std::cout << "After increment: value1 = " << ref1 << ", value2 = " << value2 << std::endl;

    return 0;
}
